daemon/algod/api/server/v2/dryrun.go

```diff
@@ -136,7 +136,7 @@ func (ddr *dryrunDebugReceiver) updateScratch() {
 
 	if any {
 		if ddr.scratchActive == nil {
-			ddr.scratchActive = make([]bool, maxActive+1, 256)
+			ddr.scratchActive = make([]bool, maxActive, 256)
```

Comment:
This doesn't seem right.  If scratch slot 1 is active, maxActive will be 1, and you need a 2 element slice, no?

Reply:
this is only called when its the first time scratch space is used and immediately after this we iterate from len(scratchActive) to maxActive inclusive using append to set the values. 

I think we could probably just create it with 0 len and capacity of 256

Reply:
We could also just make it with len of maxActive+1 and iterate from 0 to maxActive inclusive and set the values with index instead of using append